openforcefield/typing/engines/smirnoff/parameters.py

```diff
@@ -2353,6 +2713,32 @@ def create_force(self, system, topology, **kwargs):
         else:
             force = existing[0]
 
+        # Do not trust previously-calculated partial bond orders, since we don't know
+        # what method was used to assign them
+        # TODO: Jeff tried implementing a way to mark how bond orders were assigned on the
+        # topology, but realized that there's already a hierarchy of assignment
+        # methods. That is, if a molecule was assigned using PBOs_from_mols, then
+        # a different fractional bondorder method SHOULD NOT attempt
+        # recalculation, whereas if the previous method was simply DIFFERENT,
+        # then the old results should be erased/cached and overwritten with the
+        # new ones. It will be easier to handle this at the level of caching
+        # the results of molecule.assign_fractional_bond_orders
+        for top_bond in topology.topology_bonds:
+            top_bond.bond.fractional_bond_order = None
```

Comment:
Just to check my understanding: we're removing all of the bond orders in the toplogy, but `partial_bond_orders_from_molecules` will work since that argument takes in different objects? And there's enough deepcopying happening that we don't risk clearing `partial_bond_orders_from_molecules` objects here?

Reply:
Good question. Important implementation points are:
* The Topology is deepcopy-ed right at the beginning of create_openmm_system, so we're not messing around with the original. 
* Right now, `partial_bond_order_from_molecules` ("PBO mols") is a free-standing list of molecules in kwargs. So, when we wipe the partial bond orders on the `Topology`,  PBO mols aren't affected. We do a _little_ bit of redundant work by re-assigning them for each handler that may need the info from the PBO mols (meaning more isomorphism checks), but this shouldn't be a huge deal.